invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/components/DeleteRecordModal/DeleteRecordModal.js

```diff
@@ -67,38 +87,55 @@ export default class DeleteRecordModal extends Component {
     );
   };
 
+  renderAll() {
+    const { data, refProps } = this.props;
+    const canDelete = sumBy(data, 'total') === 0;
+    if (canDelete) {
+      return [this.renderDeleteHeader(), this.renderActions(canDelete)];
+    }
+
+    const modalContent = [];
+    modalContent.push(
+      refProps.map((refEntry, idx) => {
+        const refData = data[idx];
+        return [
+          this.renderHeader(refEntry, canDelete),
+          this.renderContent(refEntry, refData),
+        ];
+      })
+    );
+    modalContent.push(this.renderActions(canDelete));
+    return modalContent;
+  }
+
   render() {
-    const { isLoading, error, data, headerContent, refType } = this.props;
-    const canDelete = data.total === 0;
+    const { isLoading, data, error, refProps } = this.props;
     return (
       <Modal
         trigger={<DeleteButton onClick={this.toggleModal} />}
         open={this.state.isModalOpen}
-        onOpen={() => this.fetchReferences(this.props.checkRefs)}
+        onOpen={() =>
+          this.fetchReferences(refProps.map(entry => entry.getRefData()))
```

Comment:
To shade some light in this one, my intention is to pass a list of promises and then the actions will fire a `Promise.all()`, what happens in getRefData is we pass a function that returns a promise. If someone can come up with a clean up, would be nice!

Reply:
I think you came up with a nice solution IMHO! 👍 

I am not sure it is taking into account the fact that `refProps` can be an empty list, in case it is optional. 
For example for the `eitems`. Just to check that.